js/apps/system/_admin/aardvark/APP/react/src/views/collections/indices/addIndex/useCreateIndex.ts

```diff
@@ -30,34 +30,24 @@ export const useCreateIndex = <
         );
       },
       async () => {
-        let result;
+        let job;
         try {
-          result = await getApiRouteForCurrentDB().post(
-            `index`,
-            {
+          const db = getCurrentDB();
+          job = await db.createJob(() =>
+            db.collection(encodedCollectionName).ensureIndex({
               ...values,
               name: values.name ? String(values.name)?.normalize() : undefined
-            },
-            `collection=${encodedCollectionName}`,
-            {
-              "x-arango-async": "store"
-            }
+            } as any)
           );
-          const syncId = result.headers["x-arango-async-id"];
-          if (result.statusCode === 202 && syncId) {
-            window.arangoHelper.addAardvarkJob({
-              id: syncId,
-              type: "index",
-              desc: "Creating Index",
-              collection: collectionId
-            });
-            handleSuccess(onCloseForm);
-          }
-          if (result.body.code === 201) {
-            handleSuccess(onCloseForm);
-          }
+          window.arangoHelper.addAardvarkJob({
+            id: job.id,
+            type: "index",
+            desc: "Creating Index",
+            collection: collectionId
+          });
+          handleSuccess(onCloseForm);
         } catch (error: any) {
```

Comment:
I think this index creation change needs to be a part of the other PR? https://github.com/arangodb/arangodb/pull/21041